packages/lib-user/dev/components/App/App.js

```diff
@@ -0,0 +1,134 @@
+import oauth from 'panoptes-client/lib/oauth.js'
+import { useEffect, useState } from 'react'
+
+import { usePanoptesUser } from '@hooks/index.js'
+import { GroupStats, UserStats } from '@components/index.js'
+
+function App ({
+  groups = null,
+  users = null
+}) {
+  const [loading, setLoading] = useState(false)
+  const [userAuth, setUserAuth] = useState(null)
+
+  const { data: user, error, isLoading: userLoading } = usePanoptesUser(oauth)
+
+  useEffect(() => {
+    async function initAuthorization () {
+      setLoading(true)
+
+      try {
+        const userAuth = await oauth.init('357ac7e0e17f6d9b05587477ca98fdb69d70181e674be8e20142e1df97a84d2d');
+        setUserAuth(userAuth);
+        setLoading(false);
+      } catch (error) {
+        console.error(error);
+        setLoading(false);
+      }
+    }
+
+    initAuthorization()
+  }, [])
+
+  const login = () => oauth.signIn(window.location.origin)
+  const logout = () => oauth.signOut().then(setUserAuth)
+
+  let content = (
+    <div>
+      <h2>Key Components - urls (zooniverse.org/...)</h2>
+      <ul>
+        <li>homepage</li>
+        <li>
+          profile page (public) - users/[login]
+          <ul>
+            <li>
+              <a href={`./?users=${user?.login ?? "[login]"}/stats`}>user stats page (private) - users/{user?.login ?? "[login]"}/stats</a>
+              <ul>
+                <li>certificate - users/{user?.login ?? "[login]"}/stats/certificate</li>
+              </ul>
+            </li>
+          </ul>
+        </li>
+        <li>
+          groups list - TBD
+          <ul>
+            <li>
+              <a href="./?groups=[user_group_id]">group stats page - groups/[id]</a>
+              <ul>
+                <li>contributors - groups/[id]/contributors</li>
+              </ul>
+            </li>
+          </ul>
+        </li>
+      </ul>
+    </div>
+  )
+
+  if (groups) {
+    const subpaths = groups.split('/')
+  
+    if (subpaths[0] === '[user_group_id]') {
+      content = <p>In the url query param <code>?groups=</code>, please replace <code>[user_group_id]</code> with a user group id.</p>
+    } else if (subpaths[1] === 'contributors') {
+      content = <p>Group contributors component goes here.</p>
+    } else {
+      const groupID = subpaths[0] || ''
+
+      content = (
+        <GroupStats
+          authClient={oauth}
+          groupID={groupID}
+        />
+      )
+    }
+  }
+
+  if (users) {
+    const subpaths = users.split('/')
```

Comment:
In what case would there be more than one user? I see this variable is grabbed from the url, but just for my own understanding, when would there be more than one user's data fetched? I might just be confused by the name of the variable.

Reply:
This is for local dev work only, won't be relevant for components imported into app-root, but I've setup for the local dev work on the user stats certificate page, which might have a local dev url like `https://localhost:8080?users=[login]/stats`.

For app-root the path zooniverse.org/users/stats will have a page with SSR layout that imports and renders the client side UserStats component from lib-user, and the path zooniverse.org/users/stats/certificate will have a page with the same SSR layout but import and render a client side UserCertificate component from lib-user.

Reply:
Ah okay thank you for the clarification!